scripts/createReleasePR.mjs

```diff
@@ -198,11 +198,73 @@ async function findForkRemote() {
   }
 }
 
+/**
+ * Find the latest major version from the upstream remote
+ * @param {string} remote - The name of the remote (e.g., 'upstream')
+ * @returns {Promise<string>} The latest major version (e.g., '7', '6', etc.)
+ */
+async function findLatestMajorVersion() {
+  try {
+    // fetch tags from the GitHub API and return the last one
+    const { data: tags } = await octokit.rest.repos.listTags({
+      owner: ORG,
+      repo: REPO,
+    });
+    const tagName = tags[0].name.trim();
+    const match = tagName.match(/^v(\d+)\.(\d+)\.(\d+)/);
+    if (!match) {
+      console.error(`Error: Unable to parse version from tag ${tagName}`);
+      process.exit(1);
+    }
+    return match[1];
+  } catch (error) {
+    console.error('Error finding latest major version:', error);
+    process.exit(1);
+  }
+}
+
+/**
+ * Find the latest version for a specific major version
+ * @param majorVersion - The major version to search for (e.g., '7', '6', etc.)
+ * @returns {Promise<*|null>} - The latest tag for the specified major version, or null if not found
+ */
+async function findLastVersionForMajor(majorVersion) {
+  try {
+    const { stdout } = await execa('git', ['tag', '-l', `v${majorVersion}.*`]);
+    const tags = stdout
+      .split('\n')
+      .filter(Boolean)
+      .sort((a, b) => {
+        // Sort versions using semver logic
+        const aParts = a.substring(1).split('.').map(Number);
+        const bParts = b.substring(1).split('.').map(Number);
+
+        for (let i = 0; i < 3; i += 1) {
+          if (aParts[i] !== bParts[i]) {
+            return aParts[i] - bParts[i];
+          }
```

Comment:
This probably doesn't handle alphas, betas and rcs, does it? Maybe we can leave it for later, just wanted to point that out

Reply:
yes, it doesn't ... i'll adjust